taichi/runtime/llvm/runtime.cpp

```diff
@@ -1743,9 +1750,10 @@ i32 wasm_materialize(Context *context) {
       (RandState *)((size_t)context->runtime + sizeof(LLVMRuntime));
   // set random seed to (1, 0, 0, 0)
   context->runtime->rand_states[0].x = 1;
-  context->runtime->root =
+  // TODO: remove hard coding on root id 0(SNodeTree::kFirstID)
+  context->runtime->roots[0] =
```

Comment:
@squarefk : This could potentially break the current WASM runtime...

Reply:
I thought that it works if there's only one SNodeTree.